app/src/sharedTest/java/org/oppia/android/app/devoptions/platformparameters/PlatformParametersFragmentTest.kt

```diff
@@ -1311,24 +1270,62 @@ class PlatformParametersFragmentTest {
         atPositionOnView(
           recyclerViewId = R.id.platform_parameters_recycler_view,
           position = 0,
+          targetViewId = R.id.reset_button
+        )
+      ).perform(click())
+
+      pressBack()
+      testCoroutineDispatchers.runCurrent()
+
+      onView(withText(R.string.pending_changes_dialog_title_text))
+        .inRoot(isDialog())
+        .check(matches(isDisplayed()))
+    }
+  }
+
+  @Test
+  fun testPlatformParametersFragment_toggleParamOnAndOff_naviagteback_skipsPendingChangesDialog() {
+    setUpTestApplicationComponent()
+    launch(PlatformParametersActivity::class.java).use {
+      testCoroutineDispatchers.runCurrent()
+
+      val position = getSplashScreenWelcomeMsgPosition()
+      scrollToPosition(position)
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.platform_parameters_recycler_view,
+          position = position,
           targetViewId = R.id.platform_parameter_switch
         )
       ).perform(click())
 
-      onView(withId(R.id.save_button)).perform(click())
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.platform_parameters_recycler_view,
+          position = position,
+          targetViewId = R.id.platform_parameter_switch
+        )
+      ).perform(click())
+
+      pressBack()
       testCoroutineDispatchers.runCurrent()
 
-      launch(PlatformParametersTestActivity::class.java).use {
-        testCoroutineDispatchers.runCurrent()
-        scrollToPosition(0)
-        onView(
-          atPositionOnView(
-            recyclerViewId = R.id.platform_parameters_recycler_view,
-            position = 0,
-            targetViewId = R.id.platform_parameter_switch
-          )
-        ).check(matches(not(isChecked())))
-      }
+      onView(withText(R.string.app_restart_dialog_title))
+        .check(doesNotExist())
+    }
+  }
+
+  @Test
+  fun testPlatformParametersFragment_naviagteback_skipsPendingChangesDialog() {
```

Comment:
Also please check the spelling of "navigate" across the changes.
```suggestion
  fun testPlatformParametersFragment_navigateBackWithNoParamsModified_skipsPendingChangesDialog() {
```

Reply:
Updated